pkg/minikube/cluster/status.go

```diff
@@ -0,0 +1,510 @@
+
+	NotFound = 404
+	Stopped  = 405
+	Paused   = 418 // I'm a teapot!
+
+	// 5xx signifies a server-side error (that may be retryable)
+
+	Error               = 500
+	InsufficientStorage = 507
+	Unknown             = 520
+)
+
+var (
+	exitCodeToHTTPCode = map[int]int{
+		// exit code 26 corresponds to insufficient storage
+		26: 507,
+	}
+
+	codeNames = map[int]string{
+		100: "Starting",
+		101: "Pausing",
+		102: "Unpausing",
+		110: "Stopping",
+		103: "Deleting",
+
+		200: "OK",
+		201: "OKHAppy",
+		203: "Warning",
+		204: "Degraded",
+
+		404: "NotFound",
+		405: "Stopped",
+		418: "Paused",
+
+		500: "Error",
+		507: "InsufficientStorage",
+		520: "Unknown",
+	}
+
+	codeDetails = map[int]string{
+		507: "/var is almost out of disk space",
+	}
+)
+
+// Status holds string representations of component states
+type Status struct {
+	Name       string
+	Host       string
+	Kubelet    string
+	APIServer  string
+	Kubeconfig string
+	Worker     bool
+	TimeToStop string `json:",omitempty"`
+	DockerEnv  string `json:",omitempty"`
+	PodManEnv  string `json:",omitempty"`
+}
+
+// ClusterState holds a cluster state representation
+//
+//nolint:revive
+type ClusterState struct {
+	BaseState
+
+	BinaryVersion string
+	TimeToStop    string `json:",omitempty"`
+	Components    map[string]BaseState
+	Nodes         []NodeState
+}
+
+// NodeState holds a node state representation
+type NodeState struct {
+	BaseState
+	Components map[string]BaseState `json:",omitempty"`
+}
+
+// BaseState holds a component state representation, such as "apiserver" or "kubeconfig"
+type BaseState struct {
+	// Name is the name of the object
+	Name string
+
+	// StatusCode is an HTTP-like status code for this object
+	StatusCode int
+	// Name is a human-readable name for the status code
+	StatusName string
+	// StatusDetail is long human-readable string describing why this particular status code was chosen
+	StatusDetail string `json:",omitempty"` // Not yet implemented
+
+	// Step is which workflow step the object is at.
+	Step string `json:",omitempty"`
+	// StepDetail is a long human-readable string describing the step
+	StepDetail string `json:",omitempty"`
+}
+
+func GetStatus(api libmachine.API, cc *config.ClusterConfig) ([]*Status, error) {
+	var statuses []*Status
+	for _, n := range cc.Nodes {
+		machineName := config.MachineName(*cc, n)
+		klog.Infof("checking status of %s ...", machineName)
+		st, err := NodeStatus(api, *cc, n)
+		klog.Infof("%s status: %+v", machineName, st)
+
+		if err != nil {
+			klog.Errorf("status error: %v", err)
+			return nil, err
+		}
+		if st.Host == Nonexistent {
+			err := fmt.Errorf("the %q host does not exist", machineName)
+			klog.Errorf("%v", err)
+			return nil, err
+		}
+		statuses = append(statuses, st)
+	}
+	return statuses, nil
+}
+
+// clusterState converts Status structs into a ClusterState struct
+//
+//nolint:gocyclo
+func GetClusterState(sts []*Status, profile string, cc *config.ClusterConfig) ClusterState {
```

Comment:
```suggestion
// GetState converts Status structs into a State struct
//
//nolint:gocyclo
func GetState(sts []*Status, profile string, cc *config.ClusterConfig) State {
```

Follow up from above of stripping `Cluster` from `ClusterState`